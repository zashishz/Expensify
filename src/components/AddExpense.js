import React from 'react';

const AddExpense = () => (
    <div>
        This is Add Expense!
    </div>
);

export default AddExpense;