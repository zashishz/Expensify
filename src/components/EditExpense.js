import React from 'react';

const EditExpense = (props) => (
    <div>
        Edit Expense! {props.match.params.id}
    </div>
);

export default EditExpense;