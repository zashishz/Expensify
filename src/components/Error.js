import React from 'react';

const Error = (props) => (
    <div>
        {props.error}
    </div>
)

export default Error;