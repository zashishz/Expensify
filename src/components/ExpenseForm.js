import React from 'react';

export default class ExpenseForm extends React.Component {
    render = () => (
        <div>
            <p>Expense Form</p>
        </div>
    )
}