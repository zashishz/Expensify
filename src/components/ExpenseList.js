import React from 'react';
import { connect } from 'react-redux';
import validateExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
    <div>
        <h2>Expense List</h2>
        { 
            props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense} />
            })
        }
    </div>
);

export default connect((state) => {
    return {
        expenses: validateExpenses(state.expenses, state.filters)
    }
})(ExpenseList);