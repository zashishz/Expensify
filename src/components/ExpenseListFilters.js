import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {

    state = {
        calendarFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }

    onTextChange = (e) => {
        this.props.setTextFilter( e.target.value );
    }

    onSortChange = (e) => { 
        this.props.e.target.value == "date" ? sortByDate() : sortByAmount();
     }


    render () {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange}/>
                <select value = { this.props.filters.sortBy } onChange={this.onSortChange}>
                    <option value="amount">Amount</option>
                    <option value="date">Date</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    startDateId={"Start"}
                    endDate={this.props.filters.endDate}
                    endDateId={"End"}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);