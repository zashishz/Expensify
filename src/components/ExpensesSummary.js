import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    expensesTotal = (expensesTotal/100).toFixed(2) + ' Rs.';

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{ expenseCount }</span> { expenseWord } totalling <span>{ expensesTotal }</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpense = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpense.length,
        expensesTotal: selectExpensesTotal(visibleExpense)
    }    
}

export default connect(mapStateToProps)(ExpensesSummary);