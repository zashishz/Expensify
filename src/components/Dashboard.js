import React from 'react';
import ExpenseList from  './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';


const Dashboard = () => (
    <div>
        This is Expesify Dashboard!
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default Dashboard;