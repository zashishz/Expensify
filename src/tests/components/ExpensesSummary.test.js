import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should correctly render expenses summary with 1 expense' ,() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2000} />);
    expect(wrapper).toMatchSnapshot();
})
test('should correctly render expenses summary with multiple expense' ,() => {
    const wrapper = shallow(<ExpensesSummary expenseCount={7} expensesTotal={2892334} />);
    expect(wrapper).toMatchSnapshot();
})