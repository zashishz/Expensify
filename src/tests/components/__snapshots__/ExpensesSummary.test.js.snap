// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    20.00 Rs.
  </h2>
</div>
`;

exports[`should correctly render expenses summary with multiple expense 1`] = `
<div>
  <h2>
    Viewing 
    7
     
    expenses
     totalling 
    28923.34 Rs.
  </h2>
</div>
`;
